### DISCUSSION TITLE
How is the URL build in got? Part 3

### DISCUSSION THREAD

**Discussion by mistercorea** [sindresorhus/got#1607] [Q&A] [answered]
#### Describe the bug
https://github.com/sindresorhus/got/issues/1546 continue from 
 - Node.js version: v12.16.2
 - OS & version: OSX 10.15.5

#### Actual behavior
http://somehost/?q=%B4%BA%BD%BA
invalid url

#### Expected behavior
http://somehost/?q=%B4%BA%BD%BA
request it as it is.
...

#### Code to reproduce

```js
const got = require('got');
(async () => {
  a = await got(
    'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=%B4%BA%BD%BA&oquery=%ED%95%9C%EA%B8%80+%EC%9D%B8%EC%BD%94%EB%94%A9&tqi=huRGOwprvhGssA7smnNssssstrs-353069'
  );
})();
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by mistercorea** [comment]
source/core/index.ts:1453
```javascript
if (this.options && this.options.skipDecodeURI === true) {
  // if skipDecodeURI is set to true then do not decode it will avoid some decoding errors
} else {
  decodeURI(this.requestUrl);
}
```


```javascript
const got = require('got');
(async () => {
  a = await got(
    'https://search.naver.com/search.naver?sm=tab_hty.top&where=nexearch&query=%B4%BA%BD%BA&oquery=%ED%95%9C%EA%B8%80+%EC%9D%B8%EC%BD%94%EB%94%A9&tqi=huRGOwprvhGssA7smnNssssstrs-353069',
    { skipDecodeURI: true }
  );
})();
```
that solves the problem.
Why is it even running the decodeURI in the first place if the URL is not being assigned back to URL? if URL validation needs to run then it should run outside of got.

**Comment by szmarczak** [comment, marked as answer]
What about the tests? https://github.com/sindresorhus/got/blob/d0cd709fdd61d4e5762866309ede1b25cdbe8e60/test/arguments.ts#L38-L45

https://url.spec.whatwg.org/#percent-encoded-bytes

**Comment by szmarczak** [comment]
Got is just an HTTP client, the answer is in the specification, you should look there instead unless you find something against the spec.

**Comment by mistercorea** [comment]
hmmm I understand the spec but this problem is more for the language specific. when korean language is encoded it will encode the say above for word "news" this prevents for me to request through got. can you remove the spec check? if URL is wrong then that is up to the dev. 

if anything for validation it should have a separate lib or function.

**Comment by szmarczak** [comment]
> when korean language is encoded it will encode the say above for word "news" this prevents for me to request through got.

They way you encode it is wrong. I just checked and it's possible to encode it properly.

> can you remove the spec check? if URL is wrong then that is up to the dev.

No. Got is a spec-compliant client.
